dbms/src/Storages/KVStore/Utils/SerializationHelper.h

```diff
@@ -72,18 +80,25 @@ inline std::string readBinary2<std::string>(ReadBuffer & buf)
             "Too large string size, size={} max_size={}",
             size,
             DEFAULT_MAX_STRING_SIZE);
-    std::string s;
-    s.resize(size);
-    buf.readStrict(&s[0], size);
-    return s;
+
+    return readStringWithLength(buf, size);
 }
 
-inline std::string readStringWithLength(ReadBuffer & buf, size_t length)
+inline std::string readTiKVStringBinary(ReadBuffer & buf)
 {
-    std::string s;
-    s.resize(length);
-    buf.readStrict(&s[0], length);
-    return s;
+    UInt32 size = 0;
+    readIntBinary(size, buf);
+
+    // According to `txn-entry-size-limit` in TiDB, the max size of a TiKV key/value is 120MB.
+    // https://docs.pingcap.com/tidb/stable/tidb-configuration-file/#txn-entry-size-limit-new-in-v4010-and-v500
```

Comment:
Maybe move this to where `TIKV_MAX_VALUE_SIZE` is defined?

Reply:
fixed